notebooks/MAST/HSC/CASJOBS/HSCV3_SMC_API/hscv3_smc_api.ipynb

```diff
@@ -0,0 +1,516 @@
+   "metadata": {},
+   "source": [
+    "### Select objects with the desired magnitudes and colors near the SMC <a name=\"summary\"></a>\n",
+    "\n",
+    "This searches the summary table for objects in a 3x3 degree box centered on the galaxy that have measurements in both ACS F555W and F814W.  It computes the V-I color and selects only objects in the range -1.5 < V-I < 1.5.  This large query ultimately returns more than 700,000 objects and takes about a minute to complete."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# save typing a quoted list of columns\n",
+    "columns = \"\"\"MatchID,MatchRA,MatchDec,CI,A_F555W,A_F814W\"\"\".split(\",\")\n",
+    "columns = [x.strip() for x in columns]\n",
+    "columns = [x for x in columns if x and not x.startswith('#')]\n",
+    "\n",
+    "# select objects with at least one ACS F555W and ACS F814W measurement\n",
+    "# and with concentration index 0.9 < CI < 1.6, consistent with point sources\n",
+    "# search a large 3x3 degree box in RA and Dec centered on the SMC\n",
+    "ddec = 1.5\n",
+    "dra = ddec/np.cos(np.radians(dec))\n",
+    "constraints = {'A_F555W_N.gte': 1, 'A_F814W_N.gte': 1, 'CI.gt': 0.9, 'CI.lt': 1.6,\n",
+    "               'MatchDec.gt': dec-ddec, 'MatchDec.lt': dec+ddec,\n",
+    "               'MatchRA.gt': ra-dra, 'MatchRA.lt': ra+dra}\n",
+    "\n",
+    "# do a search with a large number of rows allowed\n",
+    "t0 = time.time()\n",
+    "tab = hscsearch(table=\"summary\", release='v3', columns=columns, verbose=True, pagesize=2000000, format='table', **constraints)\n",
+    "\n",
+    "print(\"{:.1f} s: retrieved data and converted to {}-row astropy table\".format(time.time()-t0, len(tab)))\n",
+    "\n",
+    "# compute color column and select for objects in more limited color range\n",
+    "tab['V-I'] = tab['A_F555W'] - tab['A_F814W']\n",
+    "tab = tab[(tab['V-I'] < 1.5) & (tab['V-I'] > -1.5)]\n",
+    "print(f\"{(time.time()-t0):.1f} s: selected {len(tab)} objects with -1.5 < V-I < 1.5\")\n",
+    "\n",
+    "# clean up the output format\n",
+    "tab['A_F555W'].format = \"{:.3f}\"\n",
+    "tab['A_F814W'].format = \"{:.3f}\"\n",
+    "tab['V-I'].format = \"{:.3f}\"\n",
+    "tab['CI'].format = \"{:.3f}\"\n",
+    "tab['MatchRA'].format = \"{:.6f}\"\n",
+    "tab['MatchDec'].format = \"{:.6f}\"\n",
+    "tab"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Plot object positions on the sky <a name=\"positions\"></a>\n",
+    "\n",
+    "We mark the galaxy center as well.  These fields are sprinkled all over the galaxy (as determined by the HST proposals)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(1, (10, 10))\n",
+    "plt.plot(tab['MatchRA'], tab['MatchDec'], 'bo', markersize=1, label=f'{len(tab)} HSC measurements')\n",
```

Comment:
This plot leads to a whole bunch of points on top of each other.  I get that the point is to highlight where the fields are, but it might be worth saying that specifically after this cell.  E.g. a note that says "the points all overlap, but this shows us where the fields are at least."

Reply:
done